src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -291,24 +292,32 @@ struct Analyzer {
     return worked;
   }
 
+  std::unique_ptr<SubTypes> subTypes;
```

Comment:
Is there any reason to prefer the indirection?
```suggestion
  std::optional<SubTypes> subTypes;
```

Reply:
I used `move` for the actual construction. Is that right or is there some better C++ way?

Reply:
I don't think `move` should be necessary here since what you're assigning is already an rvalue reference (I think). Are you getting an error without it?

Reply:
It seems to build both ways.